torch/_higher_order_ops/wrap.py

```diff
@@ -25,45 +30,371 @@ def wrapper():
 
 wrap = Wrap()
 
-class WrapActivationCheckpoint(HigherOrderOperator):
+random_ops = set([
```

Comment:
Can we find a way not to hardcode these?

Reply:
As an alternative (just as an idea, not sure that this is the best approach):

Unfortunately, we don't have a systematic way to get all random ops from the `torch.*` module. But if instead you could get at the ATen ops that will be captured in our checkpointed function, then you can inspect the `tags` on each the `OpOverload` for a `nondeterminism` tag (indicated randomness, [code](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/tags.yaml#L25))

If we have already used dynamo to confirm that the function is traceable, then one option would be to `make_fx()` it and get out a graph of the aten ops.

Reply:
Update: will use Brian's mode reordering stack, and check `.is_mutable` for inplace ops and `nondeterministic_seeded` tag for random ops